### PR TITLE
Fix azure mysql deployment task docs

### PR DESCRIPTION
**Pull request by mattboothman**
The MySQL task accepts parameters that are case sensitive. The example yaml markdown was updated to reflect this. I also took the time to tidy up the arguments list, and ensure it matches the example yaml, includes expected parameter values for `taskNameSelector` and `ipDetectionMethod` and noted the case sensitivity requirement for these arguments. I clarified several arguments including `sqlAdditionalArguments` which, really, ought to be named `sqlClientAdditionalOptions`.

### CODE REVIEW

**Comment by mattboothman** [comment]
Folks, this task is currently broken for Windows agents. I'm not really sure what to change in the `sqlFile` argument because of this.

The `sqlFile` parameter takes an path that is exampled in the documentation using Windows path separators. This is supposed to be unsupported by the MySQL client, which asks users switch out `\` for either `\\`, which doesn't work, or `/` which does. But the task intervenes and converts any supplied path to Windows style paths on Windows agents, so for example `../myscripts/mysql.sql` is converted to `c:\agent\_work\n\myscripts\mysql.sql`. Strangely, the path has previously worked, but only due to the omission of a terminating semicolon at the end of the resulting statement. The semicolon was added in a fix for another issue relating to a linux agent, which broke Windows agents.

I've a bug raised over at the Azure DevOps Tasks repo to fix this.

**Comment by PRMerger7** [comment]
@mattboothman : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by ktoliver** [comment]
@RoopeshNair @juliakm @steved0x Could you review this proposed bug fix? Thanks.

**Comment by juliakm** [comment]
There are a lot of great updates here. Noting a couple of things:
- azureSubscription is an alias for ConnectedServiceName
- The capitalization changes don’t match what I see in the task code so I think we need product input on what's correct.  (https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureMysqlDeploymentV1/task.json)

@steved0x @RoopeshNair Could you look at this too? I don't think I have enough of a background with this task to approve.

**Comment by mattboothman** [comment]
> * azureSubscription is an alias for ConnectedServiceName

Could switch the yaml example to use `ConnectionServiceName` instead, and revert the argument docs change for that?

> * The capitalization changes don’t match what I see in the task code so I think we need product input on what's correct.  (https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureMysqlDeploymentV1/task.json)

I can change those back to the capitalised version and update the yaml instead? No worries.

**Comment by Kittoes0124** [comment]
Why does the task perform conversion in the first place? Windows has absolutely no problem handling forward slashes and this has been a fact since **at least** XP.

**Comment by PRMerger16** [comment]
@mattboothman : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger17** [comment]
@mattboothman : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by mattboothman** [comment]
I've updated the `SqlPath` argument with info relating to the fix for the problem described in my first comment and fixed the issues from your comment, @juliakm.

**Comment by ktoliver** [comment]
@N-Usha Could you review the latest updates to the pull request? Thanks.

**Comment by mattboothman** [comment]
Folks, let me know if there’s anything else I can update

**Comment by 20shivangi** [comment]
@N-Usha @ninallam Please review this PR for doc updates today. Thanks :)

File: docs/pipelines/tasks/deploy/azure-mysql-deployment.md [hunk 1]
```diff
@@ -28,19 +28,19 @@ Use this task to run your scripts and make changes to your Azure DB for Mysql. N
...
-|`StartIpAddress`<br/>Start IP Address|(Required) The starting IP Address of the automation agent machine pool like 196.21.30.50 .|
-|`EndIpAddress`<br/>End IP Address|(Required) The ending IP Address of the automation agent machine pool like 196.21.30.65 .|
-|`DeleteFirewallRule`<br/>Delete Rule After Task Ends|(Optional) If selected, the added exception for IP addresses of the automation agent will be removed for corresponding Azure Database for MySQL|
+|`ConnectedServiceName`<br/>Azure Subscription|(Required) This is needed to connect to your Azure account.<br><br>To configure new service connection, select the Azure subscription from the list and click 'Authorize'.<br><br>If your subscription is not listed or if you want to use an existing Service Principal, you can setup an Azure service connection using 'Add' or 'Manage' button. <br/><br>Argument alias: `AzureSubscription`|
```
Comment (ninallam) [inline docs/pipelines/tasks/deploy/azure-mysql-deployment.md:31]: @mattboothman when I validate from the task assistant this argument is not capitalized. It is azureSubscription

Comment (mattboothman) [reply docs/pipelines/tasks/deploy/azure-mysql-deployment.md:31]: I can fix this. Is there convention that should be filliwed when developing tasks?

**Comment by PRMerger17** [comment]
@mattboothman : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/pipelines/tasks/includes/yaml/AzureMysqlDeploymentV1.md [hunk 2]
```diff
@@ -12,17 +12,17 @@ ms.technology: devops-cicd-tasks
...
-    sqlPassword: 
-    #taskNameSelector: 'SqlTaskFile' # Optional. Options: sqlTaskFile, inlineSqlTask
-    #sqlFile: # Required when taskNameSelector == SqlTaskFile
-    #sqlInline: # Required when taskNameSelector == InlineSqlTask
-    #sqlAdditionalArguments: # Optional
-    #ipDetectionMethod: 'AutoDetect' # Options: autoDetect, iPAddressRange
-    #startIpAddress: # Required when ipDetectionMethod == IPAddressRange
-    #endIpAddress: # Required when ipDetectionMethod == IPAddressRange
-    #deleteFirewallRule: true # Optional
+    ConnectedServiceName: # Or alias AzureSubscription
```
Comment (20shivangi) [inline docs/pipelines/tasks/includes/yaml/AzureMysqlDeploymentV1.md:15]: here also it should be azureSubscription, right ? @ninallam

Comment (ninallam) [reply docs/pipelines/tasks/includes/yaml/AzureMysqlDeploymentV1.md:15]: Yes it should be azureSubscription

**Comment by PRMerger10** [comment]
@mattboothman : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/pipelines/tasks/includes/yaml/AzureMysqlDeploymentV1.md (hunk 2 above)
Comment (mattboothman) [reply docs/pipelines/tasks/includes/yaml/AzureMysqlDeploymentV1.md]: Updated :)

**Review by ninallam — APPROVED** [review]

**Comment by 20shivangi** [comment]
@mattboothman Our PM, Nikita has approved this PR now. I think you can go ahead with the merging of this PR if all necessary people have also give approvals.

**Comment by PRMerger13** [comment]
@ktoliver : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by ktoliver** [comment]
Thanks, all.

**Comment by mattboothman** [comment]
Thanks very much for everyone's help :-)

**Pull request merged**